pip/pip-298.md

```diff
@@ -0,0 +1,99 @@
+## Motivation
+
+In the implementation of the Pulsar Transaction, each topic is configured with a `Transaction Buffer` to prevent consumers from reading uncommitted messages, which are invisible until the transaction is committed. Transaction Buffer works with Position (maxReadPosition) and `TxnID` Set (aborts).`maxReadPosition` is a position (`ledgerId` + entry sequence ID) for which it is guaranteed all messages up to it belong to a finished transaction (committed or aborted). Any messages related to an in-flight transaction are after maxReadPosition. The broker only dispatches messages, before the maxReadPosition, to the consumers. When the broker dispatches the messages before maxReadPosition to the consumer, the messages sent by aborted transactions will get filtered by the Transaction Buffer. It implements it by storing all the aborted transaction IDs in memory.
```

Comment:
I think we do not need to explain the implementation of transaction buffer in the motivation.
We can leave more information about the concept of the Pulsar transaction in the background part.

Reply:
Sure, pls cc again